packages/mask/src/plugins/Savings/SNSAdaptor/SavingsForm.tsx

```diff
@@ -0,0 +1,165 @@
+                    value={inputAmount}
+                    onChange={(e) => setInputAmount(e.target.value)}
+                />
+                <div className={classes.inputBalance}>
+                    <Button
+                        variant="outlined"
+                        size="small"
+                        color="primary"
+                        className={classes.inputButton}
+                        onClick={(e) =>
+                            setInputAmount(tab === TabType.Deposit ? formattedBalance.toString() : protocol.balance)
+                        }>
+                        Max
+                    </Button>
+                    <div className={classes.inputCurrency}>
+                        <img src={IconURLS.eth} className={classes.inputImage} />
+                        <p>{tab === TabType.Deposit ? protocol.base : protocol.pair}</p>
+                    </div>
+                </div>
+            </div>
+
+            <div className={classes.infoRow}>
+                <Typography variant="body1" className={classes.infoRowLeft}>
+                    <img src={IconURLS.eth} className={classes.rowImage} />
+                    {protocol.pair} APY%
+                </Typography>
+                <Typography variant="body1" className={classes.infoRowRight}>
+                    {protocol.apr}%
+                </Typography>
+            </div>
+            <EthereumChainBoundary
+                chainId={targetChainId}
+                noSwitchNetworkTip
+                disablePadding
+                ActionButtonPromiseProps={{
+                    fullWidth: true,
+                    classes: { root: classes.button, disabled: classes.disabledButton },
+                    color: 'primary',
+                    style: { padding: '13px 0', marginTop: 0 },
+                }}>
+                <EthereumWalletConnectedBoundary
+                    ActionButtonProps={{ color: 'primary', classes: { root: classes.button } }}
+                    classes={{ connectWallet: classes.connectWallet, button: classes.button }}>
+                    <ActionButtonPromise
+                        fullWidth
+                        color="primary"
+                        size="large"
+                        variant="contained"
+                        init={
+                            needsSwap
+                                ? 'Swap ' + protocol.pair
+                                : validationMessage ||
+                                  (tab === TabType.Deposit ? 'Deposit ' + protocol.base : 'Withdraw ' + protocol.pair)
+                        }
+                        waiting={TabType.Deposit ? 'Processing Deposit' : 'Processing Withdrawal'}
+                        failed={t('failed')}
+                        complete={t('done')}
+                        disabled={validationMessage !== '' && !needsSwap}
+                        executor={async () => {
+                            if (tab === TabType.Deposit) {
```

Comment:
Do you think any new TabType will be added? If that's possible, you can refactor the code as:

```ts
switch (tab) {
    case TabType.Deposit:
        return ...
    case TabType.Withdraw:
        return ...
    default:
        unreachable(tab)
}
```